trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/chandler/FunctionHandler.java

```diff
@@ -716,6 +688,57 @@ private Result handleFunctionCallGivenNameAndArguments(final IDispatcher main, f
 			translatedParams.add(in.getLrValue().getValue());
 			functionCallExpressionResultBuilder.addAllExceptLrValue(in);
 		}
+		if (calleeProcCType != null && calleeProcCType.hasVarArgs()) {
+			final boolean hasUsedVarargs = calleeProcCType.getVarArgsUsage() == VarArgsUsage.USED;
+			// For varargs we need a special handling:
+			// - If the varargs are not used, we simply dispatch the arguments, without passing them to the function.
+			// - If they are used, we create a pointer for all the remaining arguments and pass them to the function.
+			final AuxVarInfo auxvarinfo = mAuxVarInfoBuilder.constructAuxVarInfo(loc,
+					mTypeHandler.constructPointerType(loc), SFO.AUXVAR.VARARGS_POINTER);
+			if (hasUsedVarargs) {
+				functionCallExpressionResultBuilder.addAuxVar(auxvarinfo);
+				functionCallExpressionResultBuilder.addDeclaration(auxvarinfo.getVarDec());
+			}
+			final CPrimitive pointerType = mExpressionTranslation.getCTypeOfPointerComponents();
+			BigInteger currentOffset = BigInteger.ZERO;
+			final List<Statement> writes = new ArrayList<>();
+			final Expression originalBase = MemoryHandler.getPointerBaseAddress(auxvarinfo.getExp(), loc);
+			final Expression originalOffset = MemoryHandler.getPointerOffset(auxvarinfo.getExp(), loc);
+			for (final ExpressionResult va : varargs) {
+				final ExpressionResult param;
+				int size;
+				if (va.getCType().getUnderlyingType() instanceof CPrimitive) {
+					// All smaller types (char, short) are promoted to int
+					param = mExprResultTransformer.doIntegerPromotion(loc, va);
+					size = mTypeSizes.getSize(((CPrimitive) param.getCType()).getType());
+				} else {
+					param = va;
+					size = mTypeSizes.getSizeOfPointer();

```

Comment:
Are primitives and pointers the only possible types?

Reply:
Okay, actually I don't need the size as integer, an expression should be sufficient.  Therefore I can use `TypeSizeAndOffsetComputer::constructBytesizeExpression` and support all possible `CType`s.

Reply:
I just did that in 407bfbf410f6773d46022c1fcf7e36ffd9da63bf, just using `MemoryHandler::calculateSizeOf`, since we already have a `MemoryHandler` in this method.

Reply:
:+1: Great. I expect that in almost all cases, this should result in literals in the Boogie code, right?

Reply:
Also, do you have a small regression test for that? E.g. by passing a struct?

Reply:
> 👍 Great. I expect that in almost all cases, this should result in literals in the Boogie code, right?

For all the previous cases (primitive, pointer) this behaves exactly the same like before. The `ExpressionFactory` ensures that e.g. the addition of two literals is a literal again (and other simplifications).

> Also, do you have a small regression test for that? E.g. by passing a struct?

I will add a test for struct.